### PR TITLE
Separate running PAL tests out to their own jobs/steps.

### PR DESCRIPTION
**Pull request by jkoritzinsky**
This helps us separate the different test legs in our product that are different. Instead of including the PAL tests in the cross-job copying and the significant logic around generating work items for the src/tests test tree in Helix, create a simple dedicated job for building and running the PAL tests on each platform.

### CODE REVIEW

File: eng/pipelines/runtime.yml [hunk 1]
```diff
@@ -62,1443 +62,21 @@ extends:
...
-      #
-      - template: /eng/pipelines/common/platform-matrix.yml
-        parameters:
-          jobTemplate: /eng/pipelines/common/global-build-job.yml
-          buildConfig: checked
-          platforms:
-          - gcc_linux_x64
-          jobParameters:
-            testGroup: innerloop
-            nameSuffix: Native_GCC
```
Comment (janvorli) [inline eng/pipelines/runtime.yml:106]: Why are so many legs  including the gcc one being removed?

Comment (jkoritzinsky) [reply eng/pipelines/runtime.yml:106]: This is temporary to make testing easier. I'll revert this before I make the PR non-draft

Comment (jkoritzinsky) [reply eng/pipelines/runtime.yml]: I've reverted this portion of the change and this PR is now ready for review.

**Review by janvorli — APPROVED** [review]
It looks good, but I'd feel more confident if someone more knowledgeble in the CI yml scripts took a look too.

**Review by trylek — APPROVED** [review]
Looks great to me, thanks Jeremy!

**Comment by trylek** [comment]
Having said that, I see that helixpublishwitharcade.proj still refers to CreatePALTestArchive at line 135, I would guess that's why multiple legs of your PR failed, that will need addressing.

**Comment by jkoritzinsky** [comment]
The new jobs all passed, so merging this in.

**Pull request merged**